pysteps/nowcasts/interface.py

```diff
@@ -54,6 +60,91 @@
 _nowcast_methods["steps"] = steps.forecast
 
 
+def discover_nowcasts():
+    """
+    Search for installed nowcasts plugins in the entrypoint 'pysteps.plugin.nowcasts'
+
+    The nowcasts method found are added to the `pysteps.nowcasts.interface_nowcasts_methods`
+    dictionary containing the available nowcasts_methods.
+    """
+
+    # The pkg resources needs to be reload to detect new packages installed during
+    # the execution of the python application. For example, when the plugins are
+    # installed during the tests
+    import pkg_resources
+
+    importlib.reload(pkg_resources)
+
+    for entry_point in pkg_resources.iter_entry_points(
+        group="pysteps.plugins.nowcasts", name=None
+    ):
+        _module = entry_point.load()
+        nowcast_module_name = entry_point.name
+
+        if nowcast_module_name not in _nowcast_methods:
+            _nowcast_methods[nowcast_module_name] = _module
+
+        else:
+            RuntimeWarning(
+                f"The Nowcasts methode '{nowcast_module_name}' is already available in"
+                "'pysteps.nowcasts._nowcasts_methods'.\n"
+                f"Skipping {entry_point.module_name}:{'.'.join(entry_point.attrs)}"
+            )
+        if hasattr(nowcasts, nowcast_module_name):
+            RuntimeWarning(
+                f"The nowcasts method '{nowcast_module_name}' is already an attribute"
+                "of 'pysteps.nowcasts'.\n"
+                f"Skipping {entry_point.module_name}:{'.'.join(entry_point.attrs)}"
+            )
+        else:
+            setattr(nowcasts, nowcast_module_name, _module)
+
+
+def nowcasts_info():
+    """Print all the available nowcast methods."""
+
+    # nowcasts methods available in the `nowcasts` package
+    available_nowcasts = [
+        attr.split(".")[0]
+        for attr in os.listdir(" ".join(nowcasts.__path__))
```

Comment:
@Loickemajou Is there a more elegant way to detect all the nowcast methods that doesn't use os.listdir?